Mathlib/MeasureTheory/Measure/Lebesgue/Basic.lean

```diff
@@ -577,6 +577,90 @@ theorem volume_regionBetween_eq_lintegral [SigmaFinite μ] (hf : AEMeasurable f
           (regionBetween_subset (AEMeasurable.mk f hf) (AEMeasurable.mk g hg) s)).symm
 #align volume_region_between_eq_lintegral volume_regionBetween_eq_lintegral
 
+lemma measure_prod_univ_eq_zero_of_measure_eq_zero {α β : Type*}
+    [MeasurableSpace α] [MeasurableSpace β] {μ : Measure α} {ν : Measure β} [SigmaFinite ν]
+    {s : Set α} (s_ae_univ : μ sᶜ = 0) :
+    μ.prod ν (s ×ˢ univ)ᶜ = 0 := by
+  convert show (μ.prod ν) (sᶜ ×ˢ (univ : Set β)) = 0 by
+    simpa only [Measure.prod_prod, mul_eq_zero] using Or.inl s_ae_univ
+  ext p
+  simp only [mem_compl_iff, mem_prod, mem_univ, and_true]
+
+lemma nullMeasurableSet_prod_univ {α β : Type*} [MeasurableSpace α] [MeasurableSpace β]
```

Comment:
Wouldn't it be more natural to have a lemma saying that, if `s` and `t` are both null-measurable, then so is their product?

Reply:
Yes, of course, sorry. I think I got lazy here because I didn't immediately find the result `(s ×ˢ t)ᶜ = (sᶜ ×ˢ univ) ∪ (univ ×ˢ tᶜ)`. I still can't find it, so I would add it as `Set.compl_prod_eq_union` or `Set.compl_prod`.